### ISSUE TITLE
Redis!

### ISSUE THREAD

**Issue by Zarel** [Zarel/Pokemon-Showdown#3341]
Redis is definitely the right in-memory database for our purposes. The problem is that we can't install Redis through `npm`. So... One of us should write a script to install Redis.

Or I guess we could make users install it themselves. But I like the script idea.

**Comment by panpawn** [comment]
What about windows support? Would we write a bash script and a batch script to support both?

**Comment by Zarel** [comment]
Yes, I'm pestering Morf about it in Dev as we speak.

**Comment by Morfent** [comment]
I'd rather there be an install script out of convenience rather than making users have to install Redis themselves. I'll write the install script once I finish setting up bhyve+qemu, since Virtualbox is rather slow on my machine

**Comment by Morfent** [comment]
Will Redis be a mandatory dependency? I was thinking of making the install script run on `npm post-install` so users don't have to remember to run it themselves

**Comment by Zarel** [comment]
Yes, the only way we can really use it is as a mandatory dependency.

**Comment by Morfent** [comment]
Perfect

**Comment by Zarel** [comment]
On second thought, I was thinking of probably adding a regular RAM store for other servers... Kind of like what we do for ladders. Would be easier than installing a true database on literally everyone.

**Comment by Morfent** [comment]
What would that entail?

**Comment by Zarel** [comment]
Mostly work on my end.

**Comment by Morfent** [comment]
I'll keep what I had written in case it ends up becoming relevant again.

**Comment by DylanSp** [comment]
Don't know if this is still being considered, but how viable would it be to suggest users run Redis through Docker?

**Comment by mia-pi-git** [comment]
> Don't know if this is still being considered, but how viable would it be to suggest users run Redis through Docker?

I don't really think we're planning to use Redis for anything anymore - sqlite / postgres are the planned backends.

**Comment by Zarel** [comment]
Yeah. I'd still prefer Redis to SQLite for performance reasons, but Redis apparently can't be included with `npm install` and I don't want to maintain too many separate database backends anyway, so we're not going to do this.

We _do_ still need to add persistent storage for battle logs though, to retain battles through restarts, and also to relieve some RAM pressure.

**Comment by mia-pi-git** [comment]
> We _do_ still need to add persistent storage for battle logs though, to retain battles through restarts, and also to relieve some RAM pressure.

_Coughs in #7874_
